app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,14 @@ def handle_social_login_flow(auth)
         redirect_to return_to + hash_params, notice: "Already linked to another account!"
       end
     else # not signed in
-      if @identity&.user.present?
+      @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : @identity.user
```

Comment:
User.find_by(email: auth["info"]["email"]) : @identity.user

i dont think we need this. explain why we needed this?

Reply:
We want to check if the user is banned before he goes through the login process. There are two ways to find the user, through the `@identity` or the auth parameters. This assignment will either assign the user to `User.where(email: auth["info"]["email"])` if it exists or `@identity.user`. Once we have the user, we can check if he was banned.

Reply:
We want to check if the user is banned before he goes through the login process. There are two ways to find the user, through the @identity or the auth parameters. This assignment will either assign the user to User.where(email: auth["info"]["email"]) if it exists or @identity.user. Once we have the user, we can check if he was banned.

Reply:
Try to understand the relationships.
A `user` has many `identity`(ies). Identity is a concept. Provider is stored inside identity. User has a field email. Provider = {Facebook, Google, Twitter, Github}. Also, if a user has an identity (or a provider) then identity has email too. So, while creating the user account we already saved the user email id from identity email id. Does this makes sense?
So, I don't think we need `@identity.user`. What do you suggest?
Maybe I am wrong. Or i am missing something important here. Think carefully. It is complex code.

Reply:
Oh no I totally understand how that works. But I need to write it that way because the `@identity` **only exists** if the user has already linked his account to oauth. If for example, a user who already has a normal account through the website decides to link it to oauth, he does not have an identity and the only way to log him in is through the `oauth` parameters.